addon/components/flexberry-file.js

```diff
@@ -1,334 +1,523 @@
     return !this.get('readonly') && this.get('showUploadButton');
   }),
 
   /**
-   * Flag: indicates whether upload button is enabled now.
+    Flag: indicates whether upload button is enabled now.
+
+    @property uploadButtonIsEnabled
+    @type Boolean
+    @readonly
    */
-  uploadButtonIsEnabled: Ember.computed('uploadIsInProgress', 'downloadIsInProgress', 'uploadData', function() {
-    var uploadIsInProgress = this.get('uploadIsInProgress');
-    var downloadIsInProgress = this.get('downloadIsInProgress');
-    var selectedFile = this.get('selectedFile');
+  uploadButtonIsEnabled: Ember.computed('uploadIsInProgress', 'uploadData', function() {
+    let uploadIsInProgress = this.get('uploadIsInProgress');
+    let selectedFile = this.get('selectedFile');
 
-    return !(uploadIsInProgress || downloadIsInProgress || Ember.isNone(selectedFile));
+    return !(uploadIsInProgress || Ember.isNone(selectedFile));
   }),
 
   /**
-   * Flag: indicates whether download button is visible now.
+    Flag: indicates whether download button is visible now.
+
+    @property downloadButtonIsVisible
+    @type Boolean
+    @readonly
    */
   downloadButtonIsVisible: Ember.computed('showDownloadButton', function() {
     // Download button is always visible (but disabled if download is not available).
     return this.get('showDownloadButton');
   }),
 
   /**
-   * Flag: indicates whether download button is enabled now.
+    Flag: indicates whether download button is enabled now.
+
+    @property downloadButtonIsEnabled
+    @type Boolean
+    @readonly
    */
-  downloadButtonIsEnabled: Ember.computed('uploadIsInProgress', 'downloadIsInProgress', 'initialValue', function() {
-    var uploadIsInProgress = this.get('uploadIsInProgress');
-    var downloadIsInProgress = this.get('downloadIsInProgress');
-    var jsonInitialValue = this.get('jsonInitialValue');
+  downloadButtonIsEnabled: Ember.computed('uploadIsInProgress', 'initialValue', function() {
+    let uploadIsInProgress = this.get('uploadIsInProgress');
+    let jsonInitialValue = this.get('jsonInitialValue');
 
-    return !(uploadIsInProgress || downloadIsInProgress || Ember.isNone(jsonInitialValue));
+    return !(uploadIsInProgress || Ember.isNone(jsonInitialValue));
   }),
 
   /**
-   * Maximum file size in bytes for uploading files.
-   * It should be greater then 0 and less or equal then APP.components.file.maxUploadFileSize from application config\environment.
-   * If null or undefined, then APP.components.file.maxUploadFileSize from application config\environment will be used.
+    Maximum file size in bytes for uploading files.
+    It should be greater then 0 and less or equal then APP.components.file.maxUploadFileSize from application config\environment.
+    If null or undefined, then APP.components.file.maxUploadFileSize from application config\environment will be used.
+
+    @property maxUploadFileSize
+    @type Number
+    @default null
```

Comment:
No. Some settings are still initializing through flexberry-base-component mechanism (from configuration/environment.js) and such setings need to be undefined in component's class.